qiskit_experiments/curve_analysis/__init__.py

```diff
@@ -10,36 +10,482 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""
+r"""
 =========================================================
 Curve Analysis (:mod:`qiskit_experiments.curve_analysis`)
 =========================================================
 
 .. currentmodule:: qiskit_experiments.curve_analysis
 
-Classes
-=======
+Curve analysis provides the analysis base class for a variety of experiments with
+a single experimental parameter sweep. This analysis subclasses can override
+several class attributes to customize the behavior from data processing to post-processing,
+including providing systematic initial guess for parameters tailred to the experiment.
+Here we describe how code developers can create new analysis inheriting from the base class.
+
+
+.. _curve_analysis_overview:
+
+Curve Analysis Overview
+=======================
+
+The base class :class:`CurveAnalysis` implements the multi-objective optimization on
```

Comment:
I wonder if this is starting to get too detailed for class level API docs and should instead be moved to its own separate User Guide on curve analysis?

Reply:
Sounds good. Do we create user guide with this PR? I've already moved some long API docs (basically how code author can override method) to here and left `:ref:` link in the original place so I cannot simply remove this. Note that this is not a class level docs, this is module docs so I think this is probably okey for now.

Reply:
Yeah this can be a separate PR, I think we should do a bit of planning on what would make some good mode detailed user guides